src/riscv/lib/src/jit/builder/instr_map.rs

```diff
@@ -0,0 +1,144 @@
+use std::fmt::Debug;
+
+use crate::jit::builder::instruction::LoweredInstruction;
+use crate::jit::builder::sequence::sequence_analysis::InstrInfo;
+use crate::machine_state::memory::Address;
+
+/// A holder of the Addresses of the instructions in a sequence.
+#[derive(Debug, Clone)]
+pub(super) struct AddrMap {
+    addresses: Vec<Address>,
+}
+
+impl AddrMap {
+    /// Create a new, empty AddrMap.
+    pub(super) fn new() -> Self {
+        Self {
+            addresses: Vec::new(),
+        }
+    }
+
+    /// Insert a new address into the map.
+    pub(super) fn insert(&mut self, address: Address) {
+        self.addresses.push(address);
+    }
+
+    /// Retrieve the address at the provided instr ID.
+    pub(super) fn get(&self, instr_id: InstrId) -> Address {
+        self.addresses[instr_id.0]
+    }
```

Comment:
Nit: "get" implies it is the primary retrieval method to me, but it is not.

```suggestion
    /// Retrieve the address at the provided instr ID.
    pub(super) fn address_of(&self, instr_id: InstrId) -> Address {
        self.addresses[instr_id.0]
    }
```